pkg/util/steps/runner.go

```diff
@@ -0,0 +1,39 @@
+package steps
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"reflect"
+	"runtime"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// friendlyName returns a "friendly" stringified name of the given func.
+func friendlyName(f interface{}) string {
+	return runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
```

Comment:
you might like to be even more friendly and strip the "-fm" suffix if it exists?